wayback-core/src/main/java/org/archive/wayback/resourceindex/RemoteResourceIndex.java

```diff
@@ -124,8 +127,11 @@ public SearchResults query(WaybackRequest wbRequest)
 		ResourceNotInArchiveException, BadQueryException,
 		AccessControlException {
 //		throw new ResourceIndexNotAvailableException("oops");
-		return urlToSearchResults(getRequestUrl(wbRequest),
+        closestGroup = new ClosestTrackingCaptureFilterGroup(wbRequest, canonicalizer);
```

Comment:
Unless I'm missing something, this is a race condition as closestGroup is effectively a global variable here. A second thread can jump in and change closestGroup to a different request before its accessed below.